torch/lib/THPP/tensors/generic/THCSTensor.cpp

```diff
@@ -0,0 +1,653 @@
+}
+
+template<>
+template<typename iterator>
+auto THCSTensor<real>::resize(const iterator& begin, const iterator& end) -> THCSTensor& {
+  throw std::runtime_error("THCSTensor::resize() not supported");
+}
+
+template<>
+auto THCSTensor<real>::set(const Tensor& src) -> THCSTensor& {
+  throw std::runtime_error("THCSTensor::set() not supported");
+}
+
+template<>
+auto THCSTensor<real>::setStorage(const Storage& storage,
+                                 ptrdiff_t storageOffset,
+                                 const long_range& size,
+                                 const long_range& stride) -> THCSTensor& {
+  throw std::runtime_error("THCSTensor::setStorage not supported");
+}
+
+template<>
+auto THCSTensor<real>::setStorage(const Storage& storage,
+                                 ptrdiff_t storageOffset,
+                                 THLongStorage *size,
+                                 THLongStorage *stride) -> THCSTensor& {
+  throw std::runtime_error("THCSTensor::setStorage not supported");
+}
+
+template<>
+auto THCSTensor<real>::narrow(const Tensor& src,
+                             int dimension,
+                             long firstIndex,
+                             long size) -> THCSTensor& {
+  throw std::runtime_error("THCSTensor::narrow not supported");
+}
+
+template<>
+auto THCSTensor<real>::select(const Tensor& src, int dimension,
+                             long sliceIndex) -> THCSTensor& {
+  throw std::runtime_error("THCSTensor::select not supported");
+}
+
+template<>
+auto THCSTensor<real>::transpose(const Tensor& src, int dimension1,
+                                int dimension2) -> THCSTensor& {
+  throw std::runtime_error("THCSTensor::transpose not supported");
+}
+
+
+template<>
+auto THCSTensor<real>::unfold(const Tensor& src, int dimension,
+                             long size, long step) ->THCSTensor& {
+  throw std::runtime_error("THCSTensor::unfold not supported");
+}
+
+template<>
+auto THCSTensor<real>::squeeze(const Tensor& src, int dimension) -> THCSTensor& {
+  throw std::runtime_error("THCSTensor::squeeze not supported");
+}
+
+template<>
+auto THCSTensor<real>::unsqueeze(const Tensor& src, int dimension) -> THCSTensor& {
+  throw std::runtime_error("THCSTensor::unsqueeze not supported");
+}
+
+#ifdef THCS_REAL_IS_HALF
+#define cast_scalar(v) THC_float2half(v)
+#define uncast_scalar(v) THC_half2float(v)
+#else
+#define cast_scalar(v) v
+#define uncast_scalar(v) v
+#endif
+
+template<>
+auto THCSTensor<real>::fill(scalar_type value) -> THCSTensor& {
+  throw std::runtime_error("THCSTensor::fill() not supported");
+}
+
+template<>
+auto THCSTensor<real>::retain() -> THCSTensor& {
+  THCSTensor_(retain)(state, tensor);
+  return *this;
+}
+
+template<>
+auto THCSTensor<real>::free() -> THCSTensor& {
+  THCSTensor_(free)(state, tensor);
+  return *this;
+}
+
+#define non_const_cast(tensor) const_cast<THCSTensor&>(dynamic_cast<const THCSTensor&>(tensor))
```

Comment:
I think we've changed that in all other classes to `const_cast`, that doesn't require removing the `const` qualifier.